include/sunadjoint/sunadjoint_checkpointscheme_basic.h

```diff
@@ -0,0 +1,70 @@
+/* -----------------------------------------------------------------
+ * SUNDIALS Copyright Start
+ * Copyright (c) 2002-2024, Lawrence Livermore National Security
+ * and Southern Methodist University.
+ * All rights reserved.
+ *
+ * See the top-level LICENSE and NOTICE files for details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ * SUNDIALS Copyright End
+ * -----------------------------------------------------------------
+ * SUNAdjointCheckpointScheme_Basic class declaration.
```

Comment:
I don't want to call it `Default` since we probably wont want it to be our default once we add more schemes. However, we can call it `Fixed` (as-in fixed interval) instead of `Basic`.